hat/assets/js/apps/Iaso/domains/registry/details.tsx

```diff
@@ -0,0 +1,141 @@
+}));
+
+export const Details: FunctionComponent<Props> = ({ router }) => {
+    const {
+        params: { orgUnitId, accountId },
+        params,
+    } = router;
+    const classes: Record<string, string> = useStyles();
+    const { formatMessage } = useSafeIntl();
+    const goBack = useGoBack(router, baseUrls.registry, { accountId });
+
+    const { data: orgUnit, isFetching } = useGetOrgUnit(orgUnitId);
+
+    const subOrgUnitTypes: OrgunitTypes = useMemo(() => {
+        const options =
+            orgUnit?.org_unit_type?.sub_unit_types.map((subType, index) => ({
+                ...subType,
+                color: getOtChipColors(index) as string,
+            })) || [];
+        return orderBy(options, [f => f.depth], ['asc']);
+    }, [orgUnit]);
+
+    const { data: childrenOrgUnits } = useGetOrgUnitsChildren(
+        orgUnitId,
+        subOrgUnitTypes,
+    );
+
+    const getEnketoUrl = useGetEnketoUrl(
+        window.location.href,
+        orgUnit?.reference_instance,
+    );
+    return (
+        <>
+            <TopBar
+                title={formatMessage(MESSAGES.title)}
+                displayBackButton
+                goBack={() => goBack()}
+            />
+            <Box className={`${classes.containerFullHeightNoTabPadded}`}>
+                {isFetching && <LoadingSpinner />}
+                <Grid container spacing={2}>
+                    <Grid item xs={12} md={5}>
+                        {orgUnit && (
+                            <WidgetPaper
+                                className={classes.paper}
+                                title={orgUnit?.name ?? ''}
+                            >
+                                <OrgUnitMap
+                                    orgUnit={orgUnit}
+                                    subOrgUnitTypes={subOrgUnitTypes}
+                                    childrenOrgUnits={childrenOrgUnits || []}
+                                />
+                            </WidgetPaper>
+                        )}
+                    </Grid>
+                    {orgUnit?.reference_instance && (
+                        <Grid
+                            item
+                            xs={12}
+                            md={7}
+                            alignItems="flex-start"
+                            container
+                        >
+                            <WidgetPaper
+                                id="form-contents"
+                                className={classes.paper}
+                                title={formatMessage(MESSAGES.submission)}
+                                IconButton={IconButton}
+                                iconButtonProps={{
+                                    onClick: () => getEnketoUrl(),
+                                    overrideIcon: EnketoIcon,
+                                    color: 'secondary',
+                                    tooltipMessage: MESSAGES.editOnEnketo,
+                                }}
```

Comment:
That icon button should be hidden if the user does not have permission to edit submissions

Reply:
I would ask @madewulf first